File: advent-of-code/day02/python/one.py
import os

input_path = os.path.join(os.path.dirname(__file__), "..", "input.txt")

def add_all_repeat_numbers_in_range() -> int:
    repeat_numbers_in_range = []
    valid_range = []

    with open(input_path, "r") as file:
        ranges = file.readline().split(",")
        for inputRange in ranges:
            start, end = inputRange.strip().split("-")
            if len(start) % 2 == 1 or len(end) % 2 == 1:
                if len(start) != len(end):
                    if len(start) % 2 == 1:
                        start = "1" + "0" * len(start)
                        valid_range.append((int(start), int(end)))
                    else:
                        end = "9" * (len(end) - 1)
                        valid_range.append((int(start), int(end)))
            else:
                valid_range.append((int(start), int(end)))

    for start, end in valid_range:
        for number in range(start, end + 1):
            len_num = len(str(number))
            first_half = str(number)[: len_num // 2]
            second_half = str(number)[len_num // 2 :]
            if first_half == second_half:
                repeat_numbers_in_range.append(number)


    sum_repeat_numbers = sum(repeat_numbers_in_range)
    print("Sum of all repeat numbers in range:", sum_repeat_numbers)
    return sum_repeat_numbers

add_all_repeat_numbers_in_range()

File: advent-of-code/day02/javascript/one.js
const fs = require("fs");
const path = require("path");

const inputPath = path.join(__dirname, "..", "input.txt");
const input = fs
  .readFileSync(inputPath, "utf-8")
  .trim()
  .split(",")
  .map((str) => {
    let [rangeStart, rangeEnd] = str.split("-");
    if (rangeStart.length % 2 === 1 || rangeEnd.length % 2 === 1) {
        if (rangeStart.length === rangeEnd.length) {
            return [];
        } else if (rangeStart.length < rangeEnd.length) {
            if (rangeStart.length % 2 === 1) {
                rangeStart = "1" + "0".repeat(rangeStart.length);
            } else {
                rangeEnd = "9".repeat(rangeEnd.length - 1);
            }
        }
    }
    return [Number(rangeStart), Number(rangeEnd)];
  });

const addAllRepeatNumbersInRange = (input) => {
    const repeatNumbersInRange = [];

    for (const [rangeStart, rangeEnd] of input) {
        if (rangeStart === undefined || rangeEnd === undefined) {
            continue;
        }
        for (let num = rangeStart; num <= rangeEnd; num++) {
            const lengthOfNum = num.toString().length;
            const [firstHalf, secondHalf] = [
                num.toString().slice(0, lengthOfNum / 2),
                num.toString().slice(lengthOfNum / 2),
            ];
            if (firstHalf === secondHalf) {
                repeatNumbersInRange.push(num);
            }
        }
    }

    const sumOfRepeatNumbersInRange = repeatNumbersInRange.reduce((acc, curr) => acc + curr, 0);
    console.log("Sum of repeat numbers in range:", sumOfRepeatNumbersInRange);
    return repeatNumbersInRange;
}

addAllRepeatNumbersInRange(input);